ietfYangDraftPull/draftPullUtility.py

```diff
@@ -108,52 +108,72 @@ def check_early_revisions(directory: str, LOGGER: logging.Logger) -> None:
-                        if revision is None:
-                            continue
-
-                    # Basic date extraction can fail if there are alphanumeric characters in the revision filename part
-                    try:
-                        year = int(revision.split('-')[0])
-                        month = int(revision.split('-')[1])
-                        day = int(revision.split('-')[2])
-                    except ValueError:
-                        # Revision contained invalid characters
-                        LOGGER.exception(f'Failed to process revision for {f2}: (rev: {revision})')
-                        continue
-                    try:
-                        revisions.append(datetime(year, month, day))
-                    except ValueError:
-                        LOGGER.exception(f'Failed to process revision for {f2}: (rev: {revision})')
-                        if month == 2 and day == 29:
-                            revisions.append(datetime(year, month, 28))
-                        else:
-                            continue
-        # Single revision...
-        if len(revisions) == 0:
+        for nested_filename in filenames:
+            if get_module_name(nested_filename) != module_name:
+                continue
+            nested_filename_revision_part = nested_filename.split(module_name)[1]
+            if not is_revision_part_valid(nested_filename_revision_part):
+                continue
+            files_to_delete.append(nested_filename)
+            revision = nested_filename_revision_part.split('.')[0].replace('@', '')
+            if revision == '':
+                yang_file_path = os.path.join(directory, nested_filename)
+                revision = get_latest_revision(os.path.abspath(yang_file_path), LOGGER)
+                if revision is None:
+                    continue
+            revision = convert_revision_to_datetime(revision)
+            if revision:
+                revisions.append(revision)
+                continue
+            # Probably revision filename contained invalid characters such as alphanumeric characters
+            LOGGER.exception(f'Failed to process revision for {nested_filename}: (rev: {revision})')
+            revisions.append(revision)
+        if len(revisions) == 0:  # Single revision...
             continue
         # Keep the latest (max) revision and delete the rest
         latest = revisions.index(max(revisions))
         files_to_delete.remove(files_to_delete[latest])
-        for fi in files_to_delete:
-            if 'iana-if-type' in fi:
+        for file_to_delete in files_to_delete:
+            if 'iana-if-type' in file_to_delete:
                 break
-            os.remove(os.path.join(directory, fi))
+            if os.path.exists((path := os.path.join(directory, file_to_delete))):
+                os.remove(path)
+                
+                
+def get_module_name(filename: str) -> str:
```

Comment:
Please, put underscore before these helper methods which are called from `check_early_revisions()` to make them only internal.
Also maybe put them at the end of the file

Reply:
I put underscores before these functions, but I don't think moving these functions at the end of the file is better, because when you check what a function does then you need to go back 100+ lines of code to find where it is used, it's not convenient